### DISCUSSION TITLE
Convert a dataframe of all float columns to a 2D xarray?

### DISCUSSION THREAD

**Discussion by johann-petrak** [pydata/xarray#7526] [Q&A] [answered]
(just started looking at xarray, sorry if I overlooked something).

Is it possible to simply convert (a selection of columns) from a dataframe to a 2D DataArray and e.g. add the index or another column as the coordinate for the first dimension and the selected column names as the coordinates for the second dimension? 
And back?

This seems to be intuitively one of the most basic and straightforward conversions (e.g. for a DF that contains a multidimensional time series) but I have not seen it mentioned anywhere.

I know I can use my own code to first create the 2d numpy array and the dataarray, but I was wondering if a utility function for this maybe exists?

**Comment by dcherian** [comment]
Does this help: https://docs.xarray.dev/en/stable/generated/xarray.Dataset.from_dataframe.html

There is an equivalent DataFrame.to_xarray that calls this function IIRC

**Reply by johann-petrak** [reply]
No, because this creates a Dataset which contains a 1D DataArray representing a 1D np array for each column,  not a DataArray that represents a 2D np array.

**Reply by keewis** [reply]
you can call [`to_array`](https://docs.xarray.dev/en/stable/generated/xarray.Dataset.to_array.html) on the dataset to combine the 1D variables into a single 2D variable:
```python
df.to_xarray().to_array()
```

**Comment by jhamman** [comment, marked as answer]
You can simply pass a Dataframe to the DataArray constructor:

```python
import xarray as xr
import sklearn.datasets as skd 

xdf, ydf = skd.load_iris(return_X_y=True, as_frame=True)
da = xr.DataArray(xdf)
```

(note: this does drop the name of the index dimensions)

Or you can convert the dataframe to a Dataset  then to a DataArray:

```python
xdf.to_xarray().to_array()
```

**Reply by johann-petrak** [reply]
Thanks! I missed the obvious: passing the DF to the constructor!
